### ISSUE TITLE
In Progress Release Page

### ISSUE THREAD

**Issue by frankwiles** [cppalliance/temp-site#617]
In progress release notes need to be displayed on the site. For the currently being developed version of Boost.

**Comment by williln** [comment]
@frankwiles Some questions for the team: 

1. This seems like this is a couple of different requests: Getting the release notes on the release details page, and adding the `develop` version as a "Version" (is that a duplicate of #386?) 
2. For the release notes, are we displaying the ones that are here, but for the currently being developed version? https://www.boost.org/users/history/version_1_83_0.html If so, where do these live to retrieve them? I'm not sure if they are in a github repo, or hosted in S3, or somewhere else. I had a look around and didn't see anything obvious. 
3. For the "currently being developed" version, is that the `develop` branch? 
4. For the currently being developed version, does it need its own page separate from the existing releases details page? 
5. Should the `develop` version appear in the drop-down on the Releases and Libraries pages?

**Comment by williln** [comment]
@sdarwin Do you know where the release notes come from, the ones we should display in the body of the page on a release details page? Is it these notes? https://www.boost.org/users/history/version_1_83_0.html If so, where are those stored? A sample path for a specific version like 1.83.0 and one for the `master` or `develop` versions would be very helpful. 

This is the screen I am referencing: https://www.preview.boost.org/releases/

**Comment by sdarwin** [comment]
https://www.preview.boost.org/releases/ links to www.boost.org/users/history/version_1_83_0.html which is problematic since  the new website shouldn't have dependencies on the old website.

> where the release notes come from

https://raw.githubusercontent.com/boostorg/website/master/feed/history/boost_1_83_0.qbk

during a sync, copy that file from the GitHub (githubusercontent.com) to the database.  

Only if it's missing or within the last couple versions. Most content would already be in the DB.  If it's already there and a few versions back, it doesn't even need to generate an API call.

> what should be displayed for `master` or `develop`

option 1:  don't display release notes. or downloads, or anything other than documentation for `master` or `develop` . They aren't releases.  

eventually, another solution will need to be implemented, where the release notes are migrated into the temp-site github repository or the database itself.

**Comment by williln** [comment]
@sdarwin I know this ticket is for the "in progress" (`master`/`develop`) release notes, but let's set that aside for a second. 

About release notes for existing full Boost releases: Will the repo https://github.com/boostorg/website/ still be used once we are using the new site? Or will the process that creates that quickbook file be added to this repo https://github.com/cppalliance/temp-site/? 

Also, from prior conversations, it sounded like Python options to convert from quickbook to asciidoc or Markdown, etc. aren't great because of some customizations that Boost has. Is that right? (cc @frankwiles)

**Comment by sdarwin** [comment]
In progress release page is here: https://www.boost.org/users/history/in_progress.html   

we should also clearly mark in-progress pages similarly.

"""
In Progress Releases
Note: The release notes on this page are for releases still under development. Please don't use this page as a source of information, it's here for development purposes only. Everything is subject to change.
"""

> Will the repo https://github.com/boostorg/website/ still be used once we are using the new site?

They will exist in-parallel for some number of months.  Eventually, in the long term, it will be archived read-only.

**Comment by frankwiles** [comment]
@sdarwin do you know of any tooling that might convert qbk to other formats so we can avoid having to write a qbk parser from scratch? Is there docs on qbk anywhere we can look at if nothing else?

**Comment by sdarwin** [comment]
- Quickbook docs: https://www.boost.org/doc/libs/1_83_0/doc/html/quickbook.html

- This may not be the answer, but just for reference, building the documentation of a boost library involves compiling and using quickbook, among other tools.  An example: search for 'quickbook" on this page:  https://github.com/boostorg/release-tools/blob/develop/build_docs/linuxdocs.sh

- search in https://github.com/boostorg/website. References to quickbook.

- A boostorg/website CI file previously downloaded a quickbook release. https://github.com/boostorg/quickbook/releases    Perhaps those quickbook releases could be updated.

- Alternatives: ask boost developers to submit release notes in AsciiDoc format instead.

**Comment by sdarwin** [comment]
>  avoid having to write a qbk parser

Only speculative, but to avoid quickbook entirely:

- In future releases, after 1.84.0, request that developers submit pull requests to the new website using another markdown format which is not quickbook.

- For past releases, find a way to ingest the already-rendered release notes (html format, not qbk format) 
 https://raw.githubusercontent.com/boostorg/website/master/users/history/version_1_82_0.html
 https://raw.githubusercontent.com/boostorg/website/master/users/history/version_1_83_0.html

**Comment by williln** [comment]
> For past releases, find a way to ingest the already-rendered release notes (html format, not qbk format) 

I really like this idea and am going to go with it, and not write a parser from scratch :) 

Once that is done, I'll touch base with you @sdarwin about future releases, since that sounds like something that might require consensus.

**Comment by frankwiles** [comment]
@williln we can assume that all future versions will be in AsciiDoc, so if we can quick/hacky convert the old HTML into the system (either raw HTML or into AsciiDoc in the DB) and assume the future ones are AsciiDoc that is fine.
